dali/python/nvidia/dali/ops.py

```diff
@@ -1242,11 +1242,11 @@ def _is_real_like(input):
 
 def _to_type_desc(input):
     """ <type> description required by ArithmeticGenericOp """
-    if type(input) == bool:
+    if isinstance(input, bool):
         return "bool"
-    if type(input) == int:
+    if isinstance(input, int):
         return "int32"
-    if type(input) == float:
+    if isinstance(input, float):
```

Comment:
To keep the semantics exact, `type(x) is y` would be better than `isinstance`, which is also true for derived types.

Reply:
fixed